stream/stream.go

```diff
@@ -8,6 +8,186 @@ import (
 type Event struct {
 	Ctx        context.Context
 	Key, Value []byte
+	Ack        func()
+}
+
+type Publisher interface {
+	Subscribe(context.Context, Subscriber, ...SubscriptionOption)
+}
+
+type Subscriber interface {
+	OnNext(*Event) error
+	OnError(error) // called at most once
+	OnComplete()   // called at most once if the subscription is not canceled before the stream completes
+}
+
+type SubscriptionOption func(options SubscriptionConfig)
+
+type subscriber struct {
+	onNext     func(*Event) error
+	onError    func(error)
+	onComplete func()
+}
+
+func (s subscriber) OnNext(event *Event) error {
+	return s.onNext(event)
+}
+
+func (s subscriber) OnError(err error) {
+	s.onError(err)
+}
+
+func (s subscriber) OnComplete() {
+	s.onComplete()
+}
+
+func CreateSubscriber(onNext func(*Event) error, onError func(error), onComplete func()) Subscriber {
```

Comment:
What's the goal of this function?
There is a go proverb saying Accept interfaces, return concrete types

I don't see what this CreateSubscriber would return. A subscriber for Gorillaz stream?
If yes, why can't we have an type GorillazSubscriber that implements Subscriber?